app/services/affiliate-slots.js

```diff
@@ -81,133 +89,135 @@ const checkMobileSystem = (unit) => {
-   *
    * @param {string} query
-   * @returns {AffiliateUnit[]}
-   */
-  getAllUnitsOnSearch(query) {
-    return this.getAllUnits([
-      t => !t.disableOnSearch,
-      t => checkFilter(t.query, query),
-    ]).filter(
-      u => !u.disableOnSearch,
-    );
-  },
-
-  /**
-   * Get one BIG unit that can be displayed on current wiki with given `title`
-   *
-   * @param {string} title
-   * @returns {AffiliateUnit|undefined}
-   */
-  getBigUnitOnPage(title) {
-    const allUnits = this.getAllUnitsOnPage(title)
-      .filter(u => u.isBig);
-
-    return allUnits.length > 0 ? allUnits[0] : undefined;
-  },
-
-  /**
-   * Get one SMALL unit that can be displayed on current wiki with given `title`
-   *
-   * @param {string} title
-   * @returns {AffiliateUnit|undefined}
-   */
-  getSmallUnitOnPage(title) {
-    const allUnits = this.getAllUnitsOnPage(title)
-      .filter(u => !u.isBig);
-
-    return allUnits.length > 0 ? allUnits[0] : undefined;
-  },
-
-  /**
-   * Get all units that can be displayed on current wiki with given `title`
-   *
-   * @param {string} title
-   * @returns {AffiliateUnit[]}
-   */
-  getAllUnitsOnPage(title) {
-    return this.getAllUnits([
-      t => !t.disableOnPage,
-      t => checkFilter(t.page, title),
-    ]).filter(
-      u => !u.disableOnPage,
-    );
-  },
-
-  /**
-   * Get all units that can be displayed on current wiki
-   *
-   * @param {Function[]} filter functions for targeting
-   * @returns {AffiliateUnit[]}
+   * @returns {Targeting[]}
    */
-  getAllUnits(filters = []) {
-    let activeTargeting = targeting
-      // filter targeting by GEO cookie (country)
-      .filter(t => checkFilter(t.country, this.currentCountry))
+  _getTargetingOnSearch(query) {
+    return searchPageTargeting
       // filter targeting by vertical name
       .filter(t => checkFilter(t.vertical, this.currentVertical))
       // filter targeting by current wiki ID
-      .filter(t => checkFilter(t.wikiId, this.currentWikiId));
+      .filter(t => checkFilter(t.wikiId, this.currentWikiId))
+      // filter targeting for search `query`
+      .filter(t => checkFilter(t.query, query));
+  },
 
-    // apply additional criteria
-    filters.forEach((f) => {
-      activeTargeting = activeTargeting.filter(f);
+  fetchUnitForSearch(query, isBig = false) {
+    return new Promise((resolve, reject) => {
+      // check if we have possible units (we can fail early if we don't)
+      if (!this._getAvailableUnits()) {
+        resolve(undefined);
+      }
+
+      // get the units that fulfill the targeting on search
+      const units = _getUnitsForCampaigns(this._getTargetingOnSearch(query))
+        // filter type of ad
+        .filter(u => !!u.isBig === !!isBig)
+        // filter units disabled on search page
+        .filter(u => !u.disableOnSearch)
+      
+      // fetch only the first unit if available
+      const unit = units.length > 0 ? units[0] : undefined
```

Comment:
Any reason to have this as a var rather than in the return statement?

Reply:
Only for debugging it

Reply:
removed